query-engine/query-engine/src/logger.rs

```diff
@@ -51,40 +55,34 @@ impl<'a> Logger<'a> {
         self.log_queries = log_queries;
     }
 
-    /// Enables Jaeger telemetry.
-    pub fn enable_telemetry(&mut self, enable_telemetry: bool) {
-        self.enable_telemetry = enable_telemetry;
+    pub fn enable_metrics(&mut self, metrics: MetricRegistry) {
+        self.metrics = Some(metrics);
     }
 
-    /// Sets a custom telemetry endpoint
-    pub fn telemetry_endpoint(&mut self, endpoint: &'a str) {
-        if endpoint.is_empty() {
-            self.telemetry_endpoint = None
+    pub fn setup_telemetry(&mut self, enable_telemetry: bool, endpoint: &str) {
```

Comment:
I am always wondering with these boolean parameters: would it make sense to have `if enable_telemetry { setup_telemetry(endpoint) }` in the callers rather than this? I am likely missing something subtle.